content/data/tanzu-gemfire/blog/apache-geode-metrics-wavefront.md

```diff
@@ -0,0 +1,227 @@
+session.addSubregistry(registry);
+```
+
+On stop we will unregister the meter.
+
+```java
+session.removeSubregistry(registry);
+```
+
+## What else do we need?
+We create a file called *org.apache.geode.metrics.MetricsPublishingService* and place it in *META-INF/services* in the resource directory. The contents of this file is the path to our publisher. For this implementation it will be *com.github.jhuynh1.geode.wavefront.GeodeWavefrontPublisher*. This allows Geode to discover and use the publisher when the jar is included at start up.
+
+## Creating an “uber”/shadow jar
+We’ll be using the GFSH command line interface to include our publisher jar when starting up the cluster. To simplify the number of jars needed to be included on the classpath, we can build an “uber” jar which will include the required Micrometer Wavefront dependencies with our jar. The alternative obviously would have been to include all the jars independently on the classpath.
+Modify the build to use the shadow plugin and allow it to build an “uber” jar.
+
+```groovy
+id ‘com.github.johnrengelman.shadow’ version ‘5.2.0
+```
+
+In this particular case, we know we’ll only need to add the micrometer-wavefront-registry along with our code. We don’t need to package up Apache Geode since we will be deploying this jar onto a Geode cluster. There are probably better ways to do this, but I just created a new configuration that we’ll explicitly call out which dependencies to include
+
+```groovy
+configurations {
+    uberImplementation
+}
+shadowJar {
+    project.configurations.uberImplementation.canBeResolved = true
+    configurations = [project.configurations.uberImplementation]
+}
+```
+
+And in the dependencies we end up duplicating a dependency to indicate one is for implementation and the other specifically for the uber jar
+
+```groovy
+dependencies {
+  ...
+  implementation 'io.micrometer:micrometer-registry-wavefront:latest.release'
+  uberImplementation 'io.micrometer:micrometer-registry-wavefront:latest.release'
+}
+```
+
+To build the “uber” jar use the following command
+
+```
+./gradlew shadowJar
+```
+
+![img](/images/data-blogs/tanzu-gemfire/apache-geode-metrics-wavefront/apache-geode-metrics-directory.png)
+</br>The contents will be in the build/libs directory
+
+That’s all there is to it. If you haven’t already, just follow the instructions on Build and Deploy and you’ll be able to deploy your own implementation the same way.
+
+## Grab the complete source
+
+```
+git clone https://github.com/jhuynh1/geode-wavefront-publisher.git
+```
+
+## What Else?
+Also available is a [video walkthrough](https://www.youtube.com/watch?v=BDZh-FLkDTg)
+
+Join the [Apache Geode Community](https://geode.apache.org/community/)!
+
+[How to add more metrics to Apache Geodes implementation](https://cwiki.apache.org/confluence/display/GEODE/How+to+add+a+Meter)
+
+Create and share your own MetricsPublishingService on the [user list](http://apache%20geode%20community/)
```

Comment:
This link doesn't work

Reply:
Thank you.  The link should be: https://geode.apache.org/community/